examples/src/main/scala/io/laserdisc/kafka/DangerousGDPR1.scala

```diff
@@ -0,0 +1,56 @@
+package io.laserdisc.kafka
+
+import org.apache.kafka.streams.scala.ImplicitConversions._
+import org.apache.kafka.streams.scala.Serdes._
+import org.apache.kafka.streams.scala.StreamsBuilder
+import org.apache.kafka.streams.scala.kstream._
+import org.apache.kafka.streams.{ StreamsConfig, Topology }
+import zio.UIO
+import zio.kafka.streams.KafkaStreamsTopology
+import zio.kafka.streams.testkit.ZTestTopology
+import zio.test.Assertion.equalTo
+import zio.test._
+import zio.test.environment.TestEnvironment
+
+import java.util.Properties
+
+object DangerousGDPR1 {
+  val config: Properties = {
+    val jProperties = new Properties()
+    jProperties.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092")
+    jProperties
+  }
+
+  val builder: StreamsBuilder           = new StreamsBuilder()
+  val newUsers: KStream[String, String] = builder.stream[String, String]("new-users-input")
+  val branches: Array[KStream[String, String]] =
+    newUsers.branch((k, _) => k.startsWith("u"), (k, _) => k.startsWith("p"))
+  // array is 0 based
+  val USERNAME = 1
+  val PASSWORD = 2
```

Comment:
Stupidest error is to just put the wrong index. In this case we are writing passwords instead of user names

Reply:
This simple use case should be covered by `span`, which is defined similar to `List.span`
```scala
def span(p: (K, V) => Boolean): Task[(ZKStream[K, V], ZKStream[K, V])] =
  Task.effect(stream.branch(p, !p(_, _))).map { branches =>
    (ZKStream.newInstance(branches(0)), ZKStream.newInstance(branches(1)))
  }
```
I believe you should avoid using directly `branch` exactly for this reason!

Now, if the conditions are more than one, probably a bit more thoughts on the parameters/return types would be necessary.

Reply:
This is good, but I would frame it as a special case of the more general branching process because, as you said, it needs to work for many arities (let's say up to 10 branches)